src/wp-includes/block-template-utils.php

```diff
@@ -1444,36 +1485,65 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
  * @since 6.5.0
  * @access private
  *
- * @param stdClass        $post    An object representing a template or template part
- *                                 prepared for inserting or updating the database.
- * @param WP_REST_Request $request Request object.
- * @return stdClass The updated object representing a template or template part.
+ * @param stdClass $changes  An object representing a template or template part
+ *                           prepared for inserting or updating the database.
+ * @return stdClass|WP_Error The updated object representing a template or template part.
  */
-function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
-	$filter_name = current_filter();
-	if ( ! str_starts_with( $filter_name, 'rest_pre_insert_' ) ) {
-		return $post;
-	}
-	$post_type = str_replace( 'rest_pre_insert_', '', $filter_name );
-
+function inject_ignored_hooked_blocks_metadata_attributes( $changes ) {
```

Comment:
Yeah, this is a bit unfortunate, since we only introduced this during the 6.5 cycle and are now getting rid of the second argument already. We're only using this as a filter (for the `rest_pre_insert_` hook), and we're basically able to drop the second arg, since we can infer the required information from the first arg.

How does WP's backwards-compat policy apply in this case? Do we still need to retain the second arg? If so, we should probably just add a special case that uses `$request['id']` after all.

Reply:
Generally we retain the second arg and rename it to `$deprecated`  and emit a deprecation warning if it’s used.

So something like that:

```php
/**
 * Inject ignoredHookedBlocks metadata attributes into a template or template part.
 *
 * Given an object that represents a `wp_template` or `wp_template_part` post object
 * prepared for inserting or updating the database, locate all blocks that have
 * hooked blocks, and inject a `metadata.ignoredHookedBlocks` attribute into the anchor
 * blocks to reflect the latter.
 *
 * @since 6.5.0
 * @access private
 *
 * @param stdClass $changes  An object representing a template or template part
 *                           prepared for inserting or updating the database.
 * @param WP_REST_Request $deprecated Deprecated. Not used.
 * @return stdClass|WP_Error The updated object representing a template or template part.
 */
function inject_ignored_hooked_blocks_metadata_attributes( $changes, $deprecated = null ) {
	if ( null !== $deprecated ) {
		_deprecated_argument( __FUNCTION__, '6.5.1' );
	}
}
```

Reply:
That's an elegant pattern. Thank you @swissspidy for sharing it 👍🏻

Reply:
https://github.com/WordPress/wordpress-develop/pull/6278/commits/2d01db18e4db028b5b4feb65f0b1adaca90beea0